apps/arweave/src/ar_storage_module.erl

```diff
@@ -204,3 +219,41 @@ get_all(Start, End, [StorageModule | StorageModules], FoundModules) ->
 	get_all(Start, End, StorageModules, [StorageModule | FoundModules]);
 get_all(_Start, _End, [], FoundModules) ->
 	FoundModules.
+
+sort_storage_modules_by_left_bound(StorageModules, MaybeStoreID) ->
+	lists:sort(
+		fun({BucketSize1, Bucket1, _} = M1, {BucketSize2, Bucket2, _} = M2) ->
+			Start1 = BucketSize1 * Bucket1,
+			Start2 = BucketSize2 * Bucket2,
+			case Start1 =< Start2 of
+				false ->
+					false;
+				true ->
+					case Start1 == Start2 of
+						true ->
+							StoreID1 = ar_storage_module:id(M1),
+							StoreID2 = ar_storage_module:id(M2),
+							StoreID1 == MaybeStoreID orelse StoreID2 /= MaybeStoreID;
+						false ->
+							true
+					end
+			end
+		end,
+		StorageModules
+	).
+
+get_cover2(Start, End, _StorageModules)
+		when Start >= End ->
+	[];
+get_cover2(Start, _End, [{BucketSize, Bucket, _Packing} | _StorageModules])
+		when BucketSize * Bucket > Start ->
```

Comment:
By my read this should be `BucketSize * Bucket > End` 

What am I missing?

Reply:
The modules are sorted so if the module's start is larget than Start there is no cover and we need to return not_found. There was another bug here though, [the fix](https://github.com/ArweaveTeam/arweave/pull/556/commits/277d9fd6289c0d541a561e75c29ec53cca8695e5).

Reply:
Ah right, I was flipping it in my head. `[Start...End]` has to be *fully* contained within the module (not the other way around: module fully contained within range), so if the module's range starts after `Start` then at least part of the search range is outside the module. Got it!